testing/native/run.sh

```diff
@@ -12,19 +12,54 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
-set -eu
+set -xueo pipefail
 
 SCRIPT_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null 2>&1 && pwd )"
-DOCKER_USER=${DOCKER_USER:-couchbase}
-DOCKER_TAG=${DOCKER_TAG:-v1}
-IMAGE=${IMAGE:-$DOCKER_USER/observability-stack-test:$DOCKER_TAG}
 
-SKIP_CLUSTER_CREATION=${SKIP_CLUSTER_CREATION:-no}
+export BATS_ROOT=${BATS_ROOT:-$SCRIPT_DIR/bats}
+export BATS_FILE_ROOT=$BATS_ROOT/lib/bats-file
+export BATS_SUPPORT_ROOT=$BATS_ROOT/lib/bats-support
+export BATS_ASSERT_ROOT=$BATS_ROOT/lib/bats-assert
+export BATS_DETIK_ROOT=$BATS_ROOT/lib/bats-detik
+rm -rf  "${BATS_ROOT}"
+mkdir -p "${BATS_ROOT}/lib"
 
-docker build -f "${SCRIPT_DIR}/../microlith-test/Dockerfile" -t "${IMAGE}" "${SCRIPT_DIR}/../microlith-test/"
+BATS_ASSERT_VERSION=${BATS_ASSERT_VERSION:-2.0.0}
+BATS_SUPPORT_VERSION=${BATS_SUPPORT_VERSION:-0.3.0}
+BATS_FILE_VERSION=${BATS_FILE_VERSION:-0.3.0}
+BATS_DETIK_VERSION=${BATS_DETIK_VERSION:-1.0.0}
 
-if [[ "${SKIP_CLUSTER_CREATION}" != "yes" ]]; then
-    "${SCRIPT_DIR}/../../examples/native/run.sh"
-fi
+DOWNLOAD_TEMP_DIR=$(mktemp -d)
 
-docker run --rm -it -e TEST_NATIVE=true "${IMAGE}"
+# Install BATS helpers
+pushd "${DOWNLOAD_TEMP_DIR}"
```

Comment:
I feel like installing BATS may be better suited for the action setup, rather than the runner script (or maybe having a separate `setup.sh`) - I'd like to be able to run the tests locally without the runner script messing with my BATS install.

Reply:
Yeah I think I'll move all this into a common script to run as we'll use it for the K8S one as well.